File: Day56/DSA/Most_Frequent_Element_in_an_Array.cpp
Question:-  https://www.geeksforgeeks.org/problems/most-frequent-element-in-an-array/1

Code:-

    class Solution {
      public:
        int mostFreqEle(vector<int>& arr) {
            // code here
            unordered_map<int,int> mp;
            
            for(auto i : arr)
            {
                mp[i]++;
            }
            
            int mostfrequent = 0;
            int ele = INT_MIN;
            
            for(auto i : mp)
            {
                if(i.second > mostfrequent)
                {
                    mostfrequent = i.second;
                    ele = i.first;
                }
                else if(mostfrequent == i.second && i.first > ele)
                {
                    ele = i.first;
                }
            }
            
            return ele;
        }
    };

File: Day56/DSA/Most_Frequent_Element_in_an_Array.java
Question:-  https://www.geeksforgeeks.org/problems/most-frequent-element-in-an-array/1

Code:- 

class Solution {
    public int mostFreqEle(int[] arr) {
        // code here
        HashMap<Integer,Integer> mp = new HashMap<>();
        
        for(int i : arr)
        {
            int count = mp.getOrDefault(i,0);
            mp.put(i , count + 1);
        }
        
        int maxfreq = 0;
        
        int maxele = Integer.MIN_VALUE;
        
        for(Map.Entry<Integer,Integer> ele : mp.entrySet())
        {
            if(ele.getValue() > maxfreq)
            {
                maxfreq = ele.getValue();
                maxele = ele.getKey();
            }
            else if(ele.getValue() == maxfreq && ele.getKey() > maxele)
            {
                maxele = ele.getKey();
            }
        }
        
        return maxele;
    }
}